### PR TITLE
Add snapshot deletion to Deorbiter

### PR DESCRIPTION
**Pull request by uwe-mayer**
`e2e` tests run succesfully  --> Deorbiter successfully cleans up snapshots, therefore volumes can be deleted and CleanUp is successful

### CODE REVIEW

File: pkg/controller/deorbit/deorbiter.go [hunk 1]
```diff
@@ -138,6 +170,20 @@ func (d *ConcreteDeorbiter) DeleteServices() (deleted []core_v1.Service, err err
...
+
+	if err != nil {
+		return err
+	}
+
+	if done {
+		return nil
+	}
+
+	return wait.PollUntil(PollInterval, d.isSnapshotCleanupFinished, d.Stop)
```
Comment (Nuckal777) [inline pkg/controller/deorbit/deorbiter.go:184]: Can be replaced with `wait.PollImmediateUntil()`.

File: pkg/controller/deorbit/events.go [hunk 2]
```diff
@@ -16,6 +17,20 @@ type EventingDeorbiter struct {
 	Recorder  record.EventRecorder
 }
 
+func (d *EventingDeorbiter) DeleteSnapshots() (deleted []*unstructured.Unstructured, err error) {
+	deleted, err = d.Deorbiter.DeleteSnapshots()
+
+	for i, snapshot := range deleted {
+		if err == nil || i < (len(deleted)-1) {
```
Comment (Nuckal777) [inline pkg/controller/deorbit/events.go:24]: `i` is guaranteed to not exceed `len(deleted)` by the range expression. Also err needs to be only checked once outside of the loop.

**Review by Nuckal777 — CHANGES_REQUESTED** [review]
Added my 2 cents.

File: pkg/controller/deorbit/events.go (hunk 2 above)
Comment (Nuckal777) [reply pkg/controller/deorbit/events.go:24]: Can be simplified with an early return at `err != nil`.

File: pkg/controller/deorbit/deorbiter.go (hunk 1 above)
Comment (uwe-mayer) [reply pkg/controller/deorbit/deorbiter.go]: done

File: pkg/controller/deorbit/events.go (hunk 2 above)
Comment (uwe-mayer) [reply pkg/controller/deorbit/events.go]: Hopefully this easier to grasp:
It first logs all successfully deleted objects and the failed on last if exists.

**Pull request merged**